aten/src/ATen/native/quantized/cpu/qnnpack/include/conv_utils.h

```diff
@@ -29,43 +29,46 @@ static inline size_t compute_output_dimension(
 }  // namespace
 
 struct conv_param_t {
-  const std::array<uint32_t, 2> kernel_dims; // kernel width, height
-  const std::array<uint32_t, 2> stride_dims; // subsampling width, height
-  const std::array<uint32_t, 2> dilation; // dilation width, height
-  const std::array<uint32_t, 4> padding; // input padding top, left, bottom, right
-  const std::array<uint32_t, 2> adjustment_dims; // output adjustment
+  std::array<uint32_t, 2> kernel_dims; // kernel width, height
+  std::array<uint32_t, 2> stride_dims; // subsampling width, height
+  std::array<uint32_t, 2> dilation; // dilation width, height
+  std::array<uint32_t, 4> padding; // input padding top, left, bottom, right
+  std::array<uint32_t, 2> adjustment_dims; // output adjustment
 
-  const uint32_t groups;
-  const size_t input_channels;
-  const size_t output_channels;
-  const uint8_t kernel_zero_point;
-  const float kernel_scale;
-  const uint8_t output_min;
-  const uint8_t output_max;
-  const bool transpose;
+  uint32_t groups;
+  size_t input_channels;
+  size_t output_channels;
+  uint8_t* kernel_zero_points{nullptr};
+  float* requantization_scales{nullptr};
+  uint8_t output_min;
+  uint8_t output_max;
+  bool transpose;
+  bool is_initialized;
 
   // The following are derived parameters
   enum pytorch_qnnp_ukernel_type ukernel_type; // kernel type based on input params
   size_t group_input_channels;
   size_t group_output_channels;
 
+  conv_param_t() : is_initialized(false) {}
+
```

Comment:
I second this.  It's generally considered better practice if construction and initialization are not separated to avoid having an object in an invalid state.  I assume you are doing this because you need the default constructor, in which case the [admittedly painful] solution would be to overload the constructor of all objects that rely on an instance of conv_param_t to be default initialized if possible.

Reply:
The "painful" solution is not quite possible in this scenario because we dont have all the information we need to construct this class at the time of its default instantiation. Unfortunately this is the design we have for QNNPACK integration of conv op. Let me think if we can change that.